infinitory/errors.py

```diff
@@ -0,0 +1,137 @@
+import os
+import pickle
+import sys
+import time
+
+from datetime import datetime
+
+import logging
+
+
+class ErrorParser(object):
+    def __init__(self):
+        self._reports = dict()
+        self._all_errors = []
+        self._unique_errors = []
+
+        self._logger = logging.getLogger()
+
+    def delete_report_cache(self):
+        cache_path = './cache/'
```

Comment:
Why cache these? We don't do that for anything else, and reports are more likely to change than, say, the service metadata.

This is intended to be run every 15 minutes, so perhaps half of the reports will change every run.

Finally, the cache directory needs to be somewhere else. This is intended to run from cron or some other task runner, so it should never use the working directory.

Reply:
Specifically because I think it has to reach out to each report individually, rather than grab the report as a list. If I can change the puppetdb query to grab all of the reports at once, then I wouldn't do this.  When debugging it was particularly necessary, not sure if it's necessary in production.